l1-contracts/deploy-scripts/upgrade/EcosystemUpgrade.s.sol

```diff
@@ -0,0 +1,1345 @@
+            value: 0
+        });
+        calls[2] = Call({
+            target: addresses.bridges.sharedBridgeProxy,
+            data: abi.encodeCall(Ownable2StepUpgradeable.acceptOwnership, ()),
+            value: 0
+        });
+        calls[3] = Call({
+            target: addresses.bridgehub.ctmDeploymentTrackerProxy,
+            data: abi.encodeCall(Ownable2StepUpgradeable.acceptOwnership, ()),
+            value: 0
+        });
+    }
+
+    function getOwnerAddress() public returns (address) {
+        return config.ownerAddress;
+    }
+
+    function _getFacetCutsForDeletion() internal returns (Diamond.FacetCut[] memory facetCuts) {
+        IZKChain.Facet[] memory facets = IZKChain(config.contracts.eraDiamondProxy).facets();
+
+        // Freezability does not matter when deleting, so we just put false everywhere
+        facetCuts = new Diamond.FacetCut[](facets.length);
+        for (uint i = 0; i < facets.length; i++) {
+            facetCuts[i] = Diamond.FacetCut({
+                facet: address(0),
+                action: Diamond.Action.Remove,
+                isFreezable: false,
+                selectors: facets[i].selectors
+            });
+        }
+    }
+
+    function _composeUpgradeTx() internal returns (L2CanonicalTransaction memory transaction) {
+        transaction = L2CanonicalTransaction({
+            // FIXME: dont use hardcoded values
+            txType: 254,
+            from: uint256(uint160(L2_FORCE_DEPLOYER_ADDR)),
+            to: uint256(uint160(address(L2_COMPLEX_UPGRADER_ADDR))),
+            gasLimit: 72_000_000,
+            gasPerPubdataByteLimit: 800,
+            maxFeePerGas: 0,
+            maxPriorityFeePerGas: 0,
+            paymaster: uint256(uint160(address(0))),
+            nonce: 25,
+            value: 0,
+            reserved: [uint256(0), uint256(0), uint256(0), uint256(0)],
+            // Note, that the data is empty, it will be fully composed inside the `GatewayUpgrade` contract
+            data: new bytes(0),
+            signature: new bytes(0),
+            // All factory deps should've been published before
+            factoryDeps: new uint256[](0),
+            paymasterInput: new bytes(0),
+            // Reserved dynamic type for the future use-case. Using it should be avoided,
+            // But it is still here, just in case we want to enable some additional functionality
+            reservedDynamic: new bytes(0)
+        });
+    }
+
+    function getNewProtocolVersion() public returns (uint256) {
+        return 0x1900000000;
+    }
+
+    function getOldProtocolVersion() public returns (uint256) {
+        return 0x1800000002;
+    }
+
+    function provideSetNewVersionUpgradeCall() public returns (Call[] memory calls) {
+        // Just retrieved it from the contract
+        uint256 PREVIOUS_PROTOCOL_VERSION = getOldProtocolVersion();
+        // TODO: Is it ok to provide a deadline here?
+        uint256 DEADLINE = type(uint256).max;
```

Comment:
You brought up the idea that  it should be 7 days, just noting

Reply:
yes, agree, but then it means that chains that did not prepare for the upgrade before the deadline can not be restored... we need to figure out the entire migration process first